nf_core/subworkflows/subworkflows_command.py

```diff
@@ -0,0 +1,80 @@
+import logging
+import os
+import shutil
+from pathlib import Path
+
+from nf_core.components.components_command import ComponentCommand
+
+log = logging.getLogger(__name__)
+
+
+class SubworkflowCommand(ComponentCommand):
+    """
+    Base class for the 'nf-core subworkflows' commands
+    """
+
+    def __init__(self, dir, remote_url=None, branch=None, no_pull=False, hide_progress=False):
+        self.default_subworkflows_path = Path("subworkflows", "nf-core")
+        self.default_subworkflows_tests_path = Path("tests", "subworkflows", "nf-core")
+        super().__init__("subworkflows", dir, remote_url, branch, no_pull, hide_progress)
+
+    def get_subworkflows_clone_subworkflows(self):
```

Comment:
So that will be needed if we add the `nf-core subworkflows test` command in the same style as `modules test`.
I probably could have waited to add the implementaion of that command to the commit but I was also considering trying to centralise it further.

Reply:
got it, I thought something like this was the case. I'll have a look at `subworkflows test` then.

Reply:
I wrote it during the hackathon, and could add and push it, but I didn't try to migrate the tests for it yet.

Reply:
ah, cool! push what you have and I will migrate the tests.

Reply:
Pushed. There is definitely scope to make a `ComponentTest` command I think, as the vast majority of the code is identical to modules implementation, but it would probably need more refactoring elsewhere to implement properly. I haven't added the CLI entrypoint yet either.

Reply:
Yeah, looking through it it is very similar. I will give `ComponentTest` a try. let's see how far I come.

Reply:
Any good idea where we should put shared classes? should we also mimic the python file structure, i.e. have a `components_test.py` there for components or put all in one python file?

Reply:
So again I did start to try and prototype it but realised a full refactor was going to be too big of a job. My thought was 
```
components
 | - component_command.py
   - component_list.py
   - component_test.py
```

Not every command will be need - or be possible - to be generalise in that way, but for some of the simpler ones I think it will eventually make changes/updates much simpler. As above though I think it maybe becomes easier after we first refactor shared paths? (Also selfishly I want to get the `nf-core` path hardcoding fixed ASAP haha)

Reply:
We can probably start to use `self.component_type` once that is sorted

i.e. the `get_subworkflows_clone_subworkflows` and `get_modules_clone_modules` could be replaced by something generalised like: 

```
    def get_components_clone_components(self):
        """
        Get the components repository available in a clone of nf-core/<component_type>
        """
        component_base_path = Path(self.dir, self.get_component_default_path(self.component_type))
        return [
            str(Path(dir).relative_to(component_base_path))
            for dir, _, files in os.walk(component_base_path)
            if "main.nf" in files
        ]
```